### PR TITLE
attr: Add NambooripadPartialOrder & Leq methods

### PR DESCRIPTION
**Pull request by ChristopherRussell**
The Nambooripad partial order is a generalization of the natural partial
order for inverse semigroups to the class of regular semigroups. These
functions are modelled on the fallback methods for NaturalPartialOrder
and NaturalLeqInverseSemigroup. As I do not have a current use for this,
I haven't specialized the methods for any particular representations of
regular semigroups.

The order is defined on a regular semigroup by is x <= y if the respective
R-classes satisfy R_x <= R_y, and there is an idempotent e in R_x such
that x = e * y.

### CODE REVIEW

File: doc/attr.xml [hunk 1]
```diff
@@ -1233,3 +1233,74 @@ gap> IndecomposableElements(B);
...
+<#GAPDoc Label="NambooripadPartialOrder">
+<ManSection>
+<Attr Name="NambooripadPartialOrder" Arg="S"/>
+  <Returns>The Nambooripad partial order on an regular semigroup.</Returns>
+  <Description>
+    The <E>Nambooripad partial order</E> <M>\leq</M> on an regular semigroup
+    <A>S</A> is defined by <C>s</C><M>\leq</M><C>t</C> if the principal right
+    ideal of <C>S</C> generated by <C>s</C> is contained in the principal right
+    ideal of <C>S</C> generated by <C>t</C> and there is an idempotent
+    <C>e</C> in the R-class of <C>s</C> such that <C>s</C><M>=</M><C>et</C>. The
```
Comment (james-d-mitchell) [inline doc/attr.xml]: This should be `&R;-class` so that it is displayed correctly.

File: doc/attr.xml [hunk 2]
```diff
@@ -1233,3 +1233,74 @@ gap> IndecomposableElements(B);
...
+  </Description>
+</ManSection>
+<#/GAPDoc>
+
+<#GAPDoc Label="NambooripadPartialOrder">
+<ManSection>
+<Attr Name="NambooripadPartialOrder" Arg="S"/>
+  <Returns>The Nambooripad partial order on an regular semigroup.</Returns>
+  <Description>
+    The <E>Nambooripad partial order</E> <M>\leq</M> on an regular semigroup
```
Comment (james-d-mitchell) [inline doc/attr.xml]: "an regular" -> "a regular"

File: doc/attr.xml [hunk 3]
```diff
@@ -1233,3 +1233,74 @@ gap> IndecomposableElements(B);
...
+    <A>S</A> is defined by <C>s</C><M>\leq</M><C>t</C> if the principal right
+    ideal of <C>S</C> generated by <C>s</C> is contained in the principal right
+    ideal of <C>S</C> generated by <C>t</C> and there is an idempotent
+    <C>e</C> in the R-class of <C>s</C> such that <C>s</C><M>=</M><C>et</C>. The
+    Nambooripad partial order coincides with the natural partial order when
+    considering inverse semigroups.
```
Comment (james-d-mitchell) [inline doc/attr.xml]: Add a reference to `NaturalPartialOrder` here?

File: gap/attributes/attr.gi [hunk 4]
```diff
@@ -1080,3 +1080,57 @@ function(S)
...
+[IsSemigroup],
+function(S)
+  local elts, p, func, out, i, j;
+
+  if not IsFinite(S) then
+    ErrorNoReturn("Semigroups: NambooripadPartialOrder: usage,\n",
+                  "the argument is not a finite semigroup,");
+  elif not IsRegularSemigroup(S) then
+    ErrorNoReturn("Semigroups: NambooripadPartialOrder: usage,\n",
+                  "the argument is not an inverse semigroup,");
```
Comment (james-d-mitchell) [inline gap/attributes/attr.gi]: "an inverse" -> "a regular"

File: gap/attributes/attr.gi [hunk 5]
```diff
@@ -1080,3 +1080,57 @@ function(S)
...
+function(S)
+
+  if not IsFinite(S) then
+    ErrorNoReturn("Semigroups: NambooripadLeqRegularSemigroup: usage,\n",
+                  "the argument is not a finite semigroup,");
+  fi;
+
+  if not IsRegularSemigroup(S) then
+    ErrorNoReturn("Semigroups: NambooripadLeqRegularSemigroup: usage,\n",
+                  "the argument is not an inverse semigroup,");
```
Comment (james-d-mitchell) [inline gap/attributes/attr.gi]: "an inverse" -> "a regular"

File: gap/attributes/attr.gi [hunk 6]
```diff
@@ -1080,3 +1080,57 @@ function(S)
...
+
+InstallMethod(NambooripadLeqRegularSemigroup, "for a semigroup",
+[IsSemigroup],
+function(S)
+
+  if not IsFinite(S) then
+    ErrorNoReturn("Semigroups: NambooripadLeqRegularSemigroup: usage,\n",
+                  "the argument is not a finite semigroup,");
+  fi;
+
```
Comment (james-d-mitchell) [inline gap/attributes/attr.gi]: Remove this blank line and replace the next `if not IsRegular...` by `elif not IsRegular...`

**Comment by james-d-mitchell** [comment]
One other comment would be to perhaps make an issue suggesting that specialised methods for `Nambooripad*` could be installed for special types of semigroups.

**Comment by ChristopherRussell** [comment]
Thanks @james-d-mitchell. I believe I have now fixed all these typos, added appropriate references to `NaturalPartialOrder` and delegated calculations to `NaturalPartialOrder` in the case of an inverse semigroup.

File: doc/attr.xml [hunk 7]
```diff
@@ -1233,3 +1233,74 @@ gap> IndecomposableElements(B);
...
+    <A>S</A> is defined by <C>s</C><M>\leq</M><C>t</C> if the principal right
+    ideal of <C>S</C> generated by <C>s</C> is contained in the principal right
+    ideal of <C>S</C> generated by <C>t</C> and there is an idempotent <C>e</C>
+    in the &R;-class of <C>s</C> such that <C>s</C><M>=</M><C>et</C>. The
+    Nambooripad partial order coincides with the natural partial order when
+    considering inverse semigroups <Ref Func="NaturalPartialOrder"/>.<P/>
+
+    <C>NambooripadPartialOrder</C> returns the Nambooripad partial order on the
+    regular semigroup <A>S</A> as a list of sets of positive integers where
+    entry <C>i</C> in <C>NaturalPartialOrder(<A>S</A>)</C> is the set of
```
Comment (james-d-mitchell) [inline doc/attr.xml]: `NaturalPartialOrder` -> `NambooripadPartialOrder`

**Pull request merged**